multiqc/modules/nonpareil/nonpareil.py

```diff
@@ -0,0 +1,276 @@
+""" MultiQC module to parse output from nonpareil """
+
+
+import logging
+
+from multiqc import config
+from multiqc.modules.base_module import BaseMultiqcModule, ModuleNoSamplesFound
+
+# Initialise the logger
+from multiqc.plots import linegraph
+
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    """Nonpareil analysis is split into two parts: the first (written in C++)
+    performs the subsampling, while the second (written in R) performs the
+    statistical analyses. As such, this model requires the user to post-process
+    the R object from the second part."""
+
+    def __init__(self):
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(
+            name="nonpareil",
+            anchor="nonpareil",
+            href="https://github.com/lmrodriguezr/nonpareil",
+            info="Estimate metagenomic coverage and sequence diversity ",
+            doi="10.1093/bioinformatics/btt584",
+        )
+        # Config options
+        self.plot_observed = getattr(config, "nonpareil", {}).get("plot_observed", True)
+        self.plot_model = getattr(config, "nonpareil", {}).get("plot_model", True)
+        self.disp_type = getattr(config, "nonpareil", {}).get("plot_dispersion", False)
+
+        # Read JSON file
+        self.data_by_sample = dict()
+        for f in self.find_log_files("nonpareil", filehandles=True):
+            json_parsed = self.parse_nonpareil_json(f)
+            common_samples = set(json_parsed).intersection(self.data_by_sample)
+            if common_samples:
+                log.debug(f"Duplicate sample names found ({common_samples})! Overwriting...")
+            self.data_by_sample = self.data_by_sample | json_parsed
+        # Register samples
+        for s_name in self.data_by_sample.keys():
+            self.add_data_source(f, s_name)
```

Comment:
My idea was to read in all the files and, afterwards, register all the sample that were read but I can see that I need the original file. What if there are duplicated samples? Do they also get overwritten by `add_data_source`?

Reply:
It can handle duplicated samples, so that's not a problem :) Just need to make sure to iterate over `json_parsed` and not `self.data_by_sample` in the loop for a little efficiency - pushed that change now